ast/compile.go

```diff
@@ -5,5895 +5,123 @@
-					insertRules(set, child.Values)
-				}
-				// There might still be "sub-rules" contributing key-value "overrides" for e.g. partial object rules, continue walking
-				walk(child, i+1)
-			} else {
-				return
-			}
-
-		default:
-			// This part of the ref is a dynamic term.  We can't know what it refers
-			// to and will just need to try all of the children.
-			for _, child := range node.Children {
-				if child.Hide && !opts.IncludeHiddenModules {
-					continue
-				}
-				insertRules(set, child.Values)
-				walk(child, i+1)
-			}
-		}
-	}
-
-	walk(node, 0)
-	rules := make([]*Rule, 0, len(set))
-	for rule := range set {
-		rules = append(rules, rule)
-	}
-	return rules
-}
-
-// Utility: add all rule values to the set.
-func insertRules(set map[*Rule]struct{}, rules []util.T) {
-	for _, rule := range rules {
-		set[rule.(*Rule)] = struct{}{}
-	}
-}
-
-// RuleIndex returns a RuleIndex built for the rule set referred to by path.
-// The path must refer to the rule set exactly, i.e., given a rule set at path
-// data.a.b.c.p, refs data.a.b.c.p.x and data.a.b.c would not return a
-// RuleIndex built for the rule.
-func (c *Compiler) RuleIndex(path Ref) RuleIndex {
-	r, ok := c.ruleIndices.Get(path)
-	if !ok {
-		return nil
-	}
-	return r.(RuleIndex)
-}
-
-// PassesTypeCheck determines whether the given body passes type checking
-func (c *Compiler) PassesTypeCheck(body Body) bool {
-	checker := newTypeChecker().WithSchemaSet(c.schemaSet).WithInputType(c.inputType)
-	env := c.TypeEnv
-	_, errs := checker.CheckBody(env, body)
-	return len(errs) == 0
-}
-
-// PassesTypeCheckRules determines whether the given rules passes type checking
-func (c *Compiler) PassesTypeCheckRules(rules []*Rule) Errors {
-	elems := []util.T{}
-
-	for _, rule := range rules {
-		elems = append(elems, rule)
-	}
-
-	// Load the global input schema if one was provided.
-	if c.schemaSet != nil {
-		if schema := c.schemaSet.Get(SchemaRootRef); schema != nil {
-
-			var allowNet []string
-			if c.capabilities != nil {
-				allowNet = c.capabilities.AllowNet
-			}
-
-			tpe, err := loadSchema(schema, allowNet)
-			if err != nil {
-				return Errors{NewError(TypeErr, nil, err.Error())} //nolint:govet
-			}
-			c.inputType = tpe
-		}
-	}
-
-	var as *AnnotationSet
-	if c.useTypeCheckAnnotations {
-		as = c.annotationSet
-	}
-
-	checker := newTypeChecker().WithSchemaSet(c.schemaSet).WithInputType(c.inputType)
-
-	if c.TypeEnv == nil {
-		if c.capabilities == nil {
-			c.capabilities = CapabilitiesForThisVersion()
-		}
-
-		c.builtins = make(map[string]*Builtin, len(c.capabilities.Builtins)+len(c.customBuiltins))
-
-		for _, bi := range c.capabilities.Builtins {
-			c.builtins[bi.Name] = bi
-		}
-
-		for name, bi := range c.customBuiltins {
-			c.builtins[name] = bi
-		}
-
-		c.TypeEnv = checker.Env(c.builtins)
-	}
-
-	_, errs := checker.CheckTypes(c.TypeEnv, elems, as)
-	return errs
+	return v1.NewCompiler().WithDefaultRegoVersion(DefaultRegoVersion)
```

Comment:
So, `ast.DefaultRegoVersion == ast.RegoV0` in this package. I wonder if we should use `ast.RegoV0` here instead (and other places in the v0 API), as that is a bit more explicit ..